apm-agent-plugins/apm-hibernate-search-plugin/apm-hibernate-search-plugin-5_x/src/main/java/co/elastic/apm/agent/hibernate/search/v5_x/HibernateSearch5Instrumentation.java

```diff
@@ -0,0 +1,112 @@
+/*-
+ * #%L
+ * Elastic APM Java agent
+ * %%
+ * Copyright (C) 2018 - 2019 Elastic and contributors
+ * %%
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ * #L%
+ */
+package co.elastic.apm.agent.hibernate.search.v5_x;
+
+import java.util.Collection;
+import java.util.Collections;
+
+import javax.annotation.Nullable;
+
+import co.elastic.apm.agent.bci.ElasticApmInstrumentation;
+import co.elastic.apm.agent.bci.VisibleForAdvice;
+import co.elastic.apm.agent.hibernate.search.HibernateSearchConstants;
+import co.elastic.apm.agent.impl.transaction.Span;
+import co.elastic.apm.agent.impl.transaction.TraceContextHolder;
+import net.bytebuddy.asm.Advice;
+import net.bytebuddy.description.method.MethodDescription;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+import org.hibernate.search.query.hibernate.impl.FullTextQueryImpl;
+
+import static net.bytebuddy.matcher.ElementMatchers.hasSuperType;
+import static net.bytebuddy.matcher.ElementMatchers.isInterface;
+import static net.bytebuddy.matcher.ElementMatchers.named;
+import static net.bytebuddy.matcher.ElementMatchers.not;
+
+public class HibernateSearch5Instrumentation extends ElasticApmInstrumentation {
+
+    @Override
+    public Class<?> getAdviceClass() {
+        return HibernateSearch5ExecuteAdvice.class;
+    }
+
```

Comment:
I added a prefilter with name `Query`.
**Before:**
| HibernateSearch5Instrumentation          |   1.377.372.500 |         781.000 |
**With Prefilter:**
| HibernateSearch5Instrumentation          |     117.124.600 |       1.107.700 |

Reply:
The implementation class for the interface is called `FullTextQueryImpl`. So we will not miss something.

I have now overridden the `getClassLoaderMethod`.

Reply:
If there is only one implementation possible for this interface, no need for the name pre-matcher at all, and `getTypeMatcher` should be implemented as follows:
```
return named(org.hibernate.search.query.hibernate.impl.FullTextQueryImpl);
```

But are we looking at the right interface, or should we look at `org.hibernate.query.Query`?

Reply:
In the latest version there is only one implementation left. As I remember there were at least 2 implementations in older 5.x versions which have the same class name but resided in different packages. Could check that again if we like to only match against those two.

Reply:
Well, it depends on what we say we support 🙂 
If we say we support 5.x, we need to know we do indeed. Otherwise, we will say we support 5.11.1. Obviously the former is better

Reply:
I checked 5.0.0.Final and there are 2 `FullTextQueryImpl` classes. So I would stay at the current implementation.